tests/integration/test_service_manager.py

```diff
@@ -71,9 +92,16 @@ def discovery_client(discovery_service_stub: FakeDiscoveryServiceStub) -> Discov
 
 
 @pytest.fixture
-def discovery_service_stub() -> FakeDiscoveryServiceStub:
-    """Create a FakeDiscoveryServiceStub."""
-    return FakeDiscoveryServiceStub()
+def discovery_service_stub(request) -> FakeDiscoveryServiceStub:
```

Comment:
In the context of gRPC, `def discovery_service_stub(request):` looks like you are passing a gRPC request to the stub constructor, which doesn't make any sense. 

Please add the correct type hint, `pytest.FixtureRequest`. This makes it clear what "request" is referring to, and also enables type-checking and autocomplete.

Reply:
Removed this marker since I have changed to direct test parameterization approach.